python/tests/test_tree_stats.py

```diff
@@ -1768,6 +1768,273 @@ class TestSiteDivergence(TestDivergence, MutatedTopologyExamplesMixin):
     mode = "site"
 
 
+############################################
+# Genetic relatedness
+############################################
+
+
+def site_genetic_relatedness(
+    ts, sample_sets, indexes, windows=None, span_normalise=True, proportion=True
+):
+    out = np.zeros((len(windows) - 1, len(indexes)))
+    samples = ts.samples()
+    all_samples = list({u for s in sample_sets for u in s})
+    haps = ts.genotype_matrix(isolated_as_missing=False).T
+    denom = np.ones(len(windows))
+    if proportion:
+        denom = ts.segregating_sites(
+            sample_sets=all_samples, windows=windows, mode="site"
+        )
+    alleles = np.unique(haps)
+    for j in range(len(windows) - 1):
+        begin = windows[j]
+        end = windows[j + 1]
+        site_positions = [x.position for x in ts.sites()]
+        for i, (ix, iy) in enumerate(indexes):
+            X = sample_sets[ix]
+            Y = sample_sets[iy]
+            S = 0
+            for a in alleles:
+                this_haps = haps == a
+                haps_mean = this_haps.mean(axis=0)
+                haps_centered = this_haps - haps_mean
+                for k in range(ts.num_sites):
+                    if (site_positions[k] >= begin) and (site_positions[k] < end):
+                        for x in X:
+                            x_index = np.where(samples == x)[0][0]
+                            for y in Y:
+                                y_index = np.where(samples == y)[0][0]
+                                S += (
+                                    haps_centered[x_index][k]
+                                    * haps_centered[y_index][k]
+                                    / 2
```

Comment:
I can't seem to make this factor of 2 agree with the theory (from overleaf) - I suppose this means the overleaf is wrong? Do you know where the error is?
![Screenshot from 2020-11-26 05-17-19](https://user-images.githubusercontent.com/1046249/100356092-9faea980-2fa7-11eb-9a49-e2eb67254c02.png)

Reply:
hmmm, does it come down to the factor of 2 'trickery'? So now the naive implementation reflects equation (3) rather than (4)?